### PR TITLE
feat: new release update function

### PR DESCRIPTION
**Pull request by ahmad-kemsan**

### CODE REVIEW

File: lexactivator.go [hunk 1]
```diff
@@ -11,24 +11,33 @@ package lexactivator
...
+   LA_RELEASES_ALL     uint = 1
+   LA_RELEASES_ALLOWED uint = 2
+)
+
 var licenseCallbackFuncion callbackType
 
 var releaseUpdateCallbackFuncion callbackType
 
+var newReleaseUpdateCallbackFunction releaseCallbackType
+
```
Comment (ahmad-kemsan) [inline lexactivator.go:41]: existing variable for release update has 't' missing in ...Function so should we fix that or take advantage of that mistake and name the new one as `releaseUpdateCallbackFunction`?

Setting it to `releaseCallbackFunction` also is not correct as it is about the release update.

Comment (adnan-kamili) [reply lexactivator.go:41]: use legacy for the old one

File: lexactivator.go [hunk 2]
```diff
@@ -43,6 +52,14 @@ func releaseUpdateCallbackWrapper(status int) {
 	}
 }
 
+//export newReleaseUpdateCallbackWrapper
+func newReleaseUpdateCallbackWrapper(status int, releaseJson *C.char) {
+   releaseJsonStr := ctoGoString(releaseJson)
+   if newReleaseUpdateCallbackFunction != nil {
+      newReleaseUpdateCallbackFunction(status, releaseJsonStr)
```
Comment (adnan-kamili) [inline lexactivator.go]: where are we parsing the json?

Comment (adnan-kamili) [reply lexactivator.go]: and where is the release object

File: lexactivator.go [hunk 3]
```diff
@@ -38,11 +48,21 @@ func licenseCallbackWrapper(status int) {
...
+		legacyReleaseCallbackFunction(status)
 	}
 }
 
+//export newReleaseUpdateCallbackWrapper
+func newReleaseUpdateCallbackWrapper(status int, releaseJson *C.char) {
+   releaseJsonStr := ctoGoString(releaseJson)
+   if releaseCallbackFunction != nil {
+      var release Release
+      json.Unmarshal([]byte(releaseJsonStr), &release)
```
Comment (ahmad-kemsan) [inline lexactivator.go]: I forgot to add the check for empty releaseJsonStr.Should I do as below:

```
if releaseJsonStr != "" {
         var release Release
         json.Unmarshal([]byte(releaseJsonStr), &release)
         releaseCallbackFunction(status, release)
      } else {
         releaseCallbackFunction(status, Release{})
      }
```

**Review by adnan-kamili — APPROVED** [review]

**Review by adnan-kamili — APPROVED** [review]

**Pull request merged**